x-pack/plugins/triggers_actions_ui/public/application/components/rules_delete_modal_confirmation.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiCallOut, EuiConfirmModal } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { KueryNode } from '@kbn/es-query';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '../../common/lib/kibana';
+import { bulkDeleteRules } from '../lib/rule_api';
+import {
+  cancelButtonText,
+  getConfirmButtonText,
+  getConfirmModalText,
+  getFailedNotificationText,
+  getSuccessfulNotificationText,
+} from './translations';
+
+export const RulesDeleteModalConfirmation = ({
+  idsToDelete,
+  rulesToDeleteFilter,
+  numberOfSelectedItems,
+  onDeleted,
+  onCancel,
+  onErrors,
+  showWarningText,
+  warningText,
+  setIsDeletingRules,
+}: {
+  idsToDelete: string[];
+  rulesToDeleteFilter?: KueryNode | null;
+  numberOfSelectedItems: number;
+  onDeleted: () => Promise<void>;
+  onCancel: () => void;
+  onErrors: () => Promise<void>;
+  setIsDeletingRules: (isLoading: boolean) => void;
+  showWarningText?: boolean;
+  warningText?: string;
+}) => {
+  const numberIdsToDelete = idsToDelete.length || numberOfSelectedItems;
+
+  const [deleteModalFlyoutVisible, setDeleteModalVisibility] = useState<boolean>(false);
+
+  const singleTitle = i18n.translate('xpack.triggersActionsUI.sections.rulesList.singleTitle', {
+    defaultMessage: 'rule',
+  });
+  const multipleTitle = i18n.translate('xpack.triggersActionsUI.sections.rulesList.multipleTitle', {
+    defaultMessage: 'rules',
+  });
+
+  useEffect(() => {
+    setDeleteModalVisibility(idsToDelete.length > 0 || Boolean(rulesToDeleteFilter));
+  }, [idsToDelete, rulesToDeleteFilter]);
+
+  const {
+    http,
```

Comment:
http is not really being used here, it's just to be passed by as property to another component. What about loading it in the other component where it's actually being used? I think it would make the code easier to read/understand

Reply:
I agree. That would be better. But we are passing this as a property to function(not react component). And I see only examples how I can get http kibana service from the hook, but I do not see any function I can called and get http kibana service. 
@XavierM do you know how to do it?